x-pack/elastic-agent/pkg/agent/application/upgrade/service.go

```diff
@@ -201,9 +201,7 @@ func (p *noopPidProvider) Name() string { return "noop" }
 func (p *noopPidProvider) PID(ctx context.Context) (int, error) { return 0, nil }
 
 func invokeCmd(topPath string) *exec.Cmd {
-	homeExePath := filepath.Join(topPath, agentName)
-
-	cmd := exec.Command(homeExePath, watcherSubcommand,
+	cmd := exec.Command(paths.TopBinaryPath(), watcherSubcommand,
```

Comment:
Worth a comment that this works only because the watcher gets invoked after the symlink is rotated. That could happen in upgrade.go as well right when `InvokeWatcher` is called.

https://github.com/elastic/beats/blob/25050b6437896cc1409a3a3fc71e8b9e30aad3e7/x-pack/elastic-agent/pkg/agent/application/upgrade/upgrade.go#L161-L171

I don't think we'll be changing this too much but it will save people reading the code from having to reverse engineer the intent.

Reply:
Added a comment as requested.